x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtValidateUtil.java

```diff
@@ -67,30 +67,24 @@ public static void validate(
         final List<JWK> jwks
     ) throws Exception {
         final Date now = new Date();
-        LOGGER.debug(
-            "Validating JWT, now ["
-                + now
-                + "], alg ["
-                + jwt.getHeader().getAlgorithm()
-                + "], issuer ["
-                + jwt.getJWTClaimsSet().getIssuer()
-                + "], audiences ["
-                + jwt.getJWTClaimsSet().getAudience()
-                + "], typ ["
-                + jwt.getHeader().getType()
-                + "], auth_time ["
-                + jwt.getJWTClaimsSet().getDateClaim("auth_time")
-                + "], iat ["
-                + jwt.getJWTClaimsSet().getIssueTime()
-                + "], nbf ["
-                + jwt.getJWTClaimsSet().getIssueTime()
-                + "], exp ["
-                + jwt.getJWTClaimsSet().getExpirationTime()
-                + "], kid ["
-                + jwt.getHeader().getKeyID()
-                + "], jti ["
-                + jwt.getJWTClaimsSet().getJWTID()
-        );
+
+        if (LOGGER.isDebugEnabled()) {
```

Comment:
Nit. Why wrap this debug and not others? A few others have computed parameters too, such as converting collections/arrays to CSV strings. Should they be wrapped with LOGGER.isDebugEnabled() checks too?

Reply:
If they are `debug`, then yes. Did I miss some? I can't see it.
It's not necessary for `info` and above because we assume they are on by default.

I can't recall why I decided to use `isDebugEnabled` instead of a lambda here. I think it's the large number of parameters, but I'm not sure why the lambda seemed out of place.